README.md

```diff
@@ -803,10 +809,65 @@ Thanks also to
 [@chrishtr](https://github.com/chrishtr),
 [@dvoytenko](https://github.com/dvoytenko),
 [@housseindjirdeh](https://github.com/housseindjirdeh),
-[@jakearchibald](https://github.com/jakearchibald), and
+[@jakearchibald](https://github.com/jakearchibald),
+[@matt-buland-sfdc](https://github.com/matt-buland-sfdc), and
 [@slightlyoff](https://github.com/slightlyoff)
 for their help in exploring this space and providing feedback.
 
+## Appendix: types of navigations
+
+The web platform has many ways of initiating a navigation. For the purposes of this API, and in particular the [`navigate` event](#navigation-monitoring-and-interception), the following is intended to be a comprehensive list:
+
+- Users triggering navigations via browser UI, including (but not necessarily limited to):
+  - The URL bar
+  - The back and forward buttons
+  - The reload button
+  - Bookmarks
+- `<a>` and `<area>` elements (both directly by users, and programmatically via `element.click()`)
+- `<form>` elements (both directly by users, and programmatically via `element.submit()`)
+- `<meta http-equiv="refresh">`
+- The `Refresh` HTTP response header
+- The `window.location` setter, the various `location.*` setters, and the `location.replace()`, `location.assign()`, and `location.reload()` methods
+- Calling `window.open(url, nameOfSomeWindow)` will navigate a window whose `window.name` is `nameOfSomeWindow`
+- `history.back()`, `history.forward()`, and `history.go()`
+- `history.pushState()` and `history.replaceState()`
+- `appHistory.back()`, `appHistory.forward()`, `appHistory.navigateTo()`
+- `appHistory.pushNewEntry()` and `appHistory.updateCurrentEntry()`
+- [`document.open()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/open)
+
+**Cross-document navigations** are navigations where, after the navigation completes, you end up in a different `Document` object than the one you are curently on. Notably, these unload the old document, and stop running any JavaScript code from there.
+
+**Same-document** navigations are ones where, after the navigation completes, you stay on the same `Document`, with the same JavaScript environment.
+
+Most navigations are cross-document navigations. Same-document navigations can happen due to:
```

Comment:
What can be confusing for many users is that these mechanisms _assume_ they are cross-document, even if they wind up fetching the same document at runtime. Using the wrong mechanism leads to lower page performance and excessive refreshes because they are not integrated into the SPA. I think this section does a good job of describing the expectations, and the real goal for SPAs is to get as many clicks as possible into that beloved 3rd column: `navigate` event.